### PR TITLE
Feature/162 refactor reports

### LINKED ISSUE #58: Move the "report saving" functionality from Main to the usecases package
Currently the saving of Release/Validate report is done in the main package.

This is part of the use cases, and should be done moved away from Main.

### LINKED ISSUE #162: Refactor reports
## Problem

The validation report contains duplicated lines because we changed validators' logic. We need to remove duplicated validation from the report

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #162 
Closes #58

### CODE REVIEW

File: src/main/java/com/exasol/releasedroid/main/ReleaseDroid.java [hunk 1]
```diff
@@ -67,7 +64,25 @@ public void run(final UserInput userInput) {
         } else if (userInput.getGoal() == Goal.RELEASE) {
             reports.addAll(this.releaseUseCase.release(repository, platformNames));
         }
-        writeReportToDisk(userInput, platformNames, reports);
+        sendResponse(createResponse(reports, userInput, platformNames));
+    }
+
+    private void sendResponse(final ReleaseDroidResponse response) {
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releasedroid/main/ReleaseDroid.java:70]: I find this name a bit unclear. Maybe we can rename it as `processResponse` or something more meaningful?

File: src/main/java/com/exasol/releasedroid/usecases/response/ReleaseDroidResponse.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+import com.exasol.releasedroid.usecases.request.PlatformName;
+
+/**
+ * The release droid response.
+ */
+public class ReleaseDroidResponse {
+    private final String fullRepositoryName;
+    private final Goal goal;
+    private final String branch;
+    private final String localPath;
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releasedroid/usecases/response/ReleaseDroidResponse.java:16]:
```suggestion
    private final String localRepositoryPath;
```

File: src/main/java/com/exasol/releasedroid/formatting/ReportLoggerFormatter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import com.exasol.releasedroid.usecases.report.Report;
+import com.exasol.releasedroid.usecases.report.Result;
+
+/**
+ * Formatting report for the logger output.
+ */
+public class ReportLoggerFormatter implements ReportFormatter {
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releasedroid/formatting/ReportLoggerFormatter.java:14]: I find this name a bit confusing. Maybe we can rename it?

File: src/main/java/com/exasol/releasedroid/usecases/report/AbstractReport.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            if (result.isFailed()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void merge(final Report report) {
+        if (!report.getReportName().equals(getReportName())) {
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releasedroid/usecases/report/AbstractReport.java:34]: this here is breaking the interface contract. I would suggest to rollback this change and have a single Report class

File: src/main/java/com/exasol/releasedroid/usecases/report/Result.java [hunk 5]
```diff
@@ -1,21 +1,91 @@
...
+     *
+     * @return platform names
+     */
+    public List<PlatformName> getPlatformNames() {
+        return this.platformNames;
+    }
+
+    protected void addPlatforms(final List<PlatformName> platformNames) {
+        for (final PlatformName platformName : platformNames) {
+            if (!this.platformNames.contains(platformName)) {
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releasedroid/usecases/report/Result.java:62]: maybe you can use a `Set` for this

Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/releasedroid/usecases/report/Result.java:62]: The formatting part where I print the names is not stable in tests if I use Set. As we won't ever have more than 10 platforms here, I think it's not really a big performance issue to have an O(N^2) complexity here

**Review by chiaradiamarcelo — APPROVED** [review]

**Pull request merged**